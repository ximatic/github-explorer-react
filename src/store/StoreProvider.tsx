'use client';

import { useRef } from 'react';
import { Provider } from 'react-redux';

import { AppStore, makeStore } from './store';

interface StoreProviderProps {
  readonly children: React.ReactNode;
}

export default function StoreProvider({ children }: StoreProviderProps) {
  const storeRef = useRef<AppStore | null>(null);
  if (!storeRef.current) {
    // Create the store instance the first time this renders
    storeRef.current = makeStore();
  }

  return <Provider store={storeRef.current}>{children}</Provider>;
}
