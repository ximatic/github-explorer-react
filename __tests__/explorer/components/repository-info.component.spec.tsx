import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { PrimeReactProvider } from 'primereact/api';

import StoreProvider from '../../../src/store/StoreProvider.tsx';

import { MOCK_REPOSITORY_1, MOCK_REPOSITORY_2 } from '../../../__mocks__/explorer.mocks.ts';

import RepositoryInfo from '../../../src/explorer/components/repository-info.component.tsx';

describe('Header', () => {
  test('renders repository panel with repository URL', async () => {
    const { queryAllByText } = render(
      <StoreProvider>
        <BrowserRouter>
          <PrimeReactProvider>
            <RepositoryInfo repository={MOCK_REPOSITORY_1} />
          </PrimeReactProvider>
        </BrowserRouter>
      </StoreProvider>,
    );
    expect(queryAllByText(new RegExp(`\\b(${MOCK_REPOSITORY_1.owner}/${MOCK_REPOSITORY_1.name})\\b`))).toBeTruthy();
    expect(queryAllByText(/Repository URL/i)).toBeTruthy();
  });
  test('renders repository panel without repository URL', async () => {
    const { queryAllByText } = render(
      <StoreProvider>
        <BrowserRouter>
          <PrimeReactProvider>
            <RepositoryInfo repository={MOCK_REPOSITORY_2} />
          </PrimeReactProvider>
        </BrowserRouter>
      </StoreProvider>,
    );
    expect(queryAllByText(new RegExp(`\\b(${MOCK_REPOSITORY_2.owner}/${MOCK_REPOSITORY_2.name})\\b`))).toBeTruthy();
    expect(queryAllByText(/Repository URL/i)).toEqual([]);
  });
});
