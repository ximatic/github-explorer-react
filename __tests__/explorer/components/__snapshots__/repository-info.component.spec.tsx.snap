// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="repository-info"
  >
    <div
      class="p-panel p-component"
      data-pc-name="panel"
      data-pc-section="root"
      id="pr_id_1"
    >
      <div
        class="p-panel-header"
        data-pc-section="header"
      >
        <span
          class="p-panel-title"
          data-pc-section="title"
          id="pr_id_1_header"
        >
          test-owner-1/test-name-1
        </span>
        <div
          class="p-panel-icons"
          data-pc-section="icons"
        />
      </div>
      <div
        aria-hidden="false"
        aria-labelledby="pr_id_1_header"
        class="p-toggleable-content"
        data-pc-section="toggleablecontent"
        id="pr_id_1_content"
        role="region"
      >
        <div
          class="p-panel-content"
          data-pc-section="content"
        >
          <div>
            <span
              class="pi pi-star-fill mr-2"
            />
            0 | 01-07-2025
          </div>
          <div
            class="mt-1"
          >
            test-description-1
          </div>
          <div
            class="mt-2"
          >
            <a
              class="underline"
              href="test-url-1"
              target="_blank"
            >
              <span
                class="pi pi-link mr-2"
              />
              Repository URL
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
